### DISCUSSION TITLE
Why Does `math.matrixFromFunction` Return A Vanilla Array/Matrix?

### DISCUSSION THREAD

**Discussion by fweth** [josdejong/mathjs#2910] [Q&A] [answered]
I just wonder, looking at the documentation, `math.matrixFromFunction` should return an `Array`, so a Math.js object like `math.zeros`, but it returns a vanilla JS array.  For my own usecases, I prefer vanilla JS arrays, so I'm happy with that!  I just wonder where the difference comes from...

**Comment by josdejong** [comment, marked as answer]
In general, the idea is: Array in -> Array out, Matrix in -> Matrix out.

```js
math.matrixFromFunction([3,3], i => i[0] - i[1])              // Array 
math.matrixFromFunction(math.matrix([3,3]), i => i[0] - i[1]) // Matrix
```
